Fixed a bug that CliffrodRZSetTranspiler generates T or TDag gates

lqtmirage opened the pull request:
- Fixed a bug that caused the `CliffordRZSetTranspiler` to generate a `T` / `TDag` gates
- Added option to `RZ2NamedTranspiler` to generate `T` / `TDag` gates or not

lqtmirage committed:
rz to named support only clifford conversion

lqtmirage committed:
remove unnecessary trans from clifford rz set trans

ThomasenQunasys commented:
@lqtmirage thank you Wada-san! Sorry for the late review.
I tried running this with
```
import numpy as np

from quri_parts.circuit.transpile import CliffordRZSetTranspiler
from quri_parts.circuit import QuantumCircuit
from quri_parts.circuit.utils.circuit_drawer import draw_circuit

circuit = QuantumCircuit(3)
circuit.add_H_gate(0)
circuit.add_CNOT_gate(0, 1)
circuit.add_RZ_gate(2, 0.5* np.pi)
circuit.add_RZ_gate(2, 0.25* np.pi)

transpiler = CliffordRZSetTranspiler()
transpiled_circuit = transpiler(circuit)

print("Original Circuit:")
draw_circuit(circuit)
print("\nTranspiled Circuit:")
draw_circuit(transpiled_circuit)
```
and obtained 
```
Original Circuit:
   ___          
  | H |         
--|0  |-----●---
  |___|     |   
           _|_  
          |CX | 
----------|1  |-
          |___| 
   ___     ___  
  |RZ |   |RZ | 
--|2  |---|3  |-
  |___|   |___| 

Transpiled Circuit:
   ___          
  | H |         
--|0  |-----●---
  |___|     |   
           _|_  
          |CX | 
----------|1  |-
          |___| 
   ___     ___  
  | S |   | T | 
--|2  |---|3  |-
  |___|   |___|
```
It seems like it still produces a T-gate

lqtmirage committed:
fix rz2 named generate t if it turned off, add tests

lqtmirage commented:
Thank you very much for reviewing and testing.
I added a test case, confirmed similar results, and fixed the bug. (The pattern that `RZ2NamedTranspiler` generates T gates was missed.)

ThomasenQunasys reviewed (approved):
Thanks! LGTM
